airbyte-webapp/src/core/form/types.ts

```diff
@@ -48,3 +48,6 @@ export type { FormBlock, FormConditionItem, FormGroupItem, FormObjectArrayItem }
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export type WidgetConfig = { [key: string]: any };
 export type WidgetConfigMap = { [key: string]: WidgetConfig };
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type FormComponentOverrideProps = Record<string, any>;
```

Comment:
There isn't any real need making this an `any` record and thus disabling type checking on this pretty much in places it's used. We should type this properly with all overwrites we know should be applicable (i.e. `{ disabled?: boolean }`), and can extend it if we ever need more.

Reply:
Unfortunately, the typing in this area of the code is in really bad shape. Based on this comment I documented the current issue in #13331